### ISSUE TITLE
Git unresponsive

### ISSUE THREAD

**Issue by BrentHumphreys** [microsoft/vscode#73493]
Issue Type: <b>Performance Issue</b>

Git module in VSCode shows unresponsive.

VS Code version: Code - Insiders 1.34.0-insider (473af338e1bd9ad4d9853933da1cd9d5d9e07dc9, 2019-05-01T00:22:05.899Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2712)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.92GB (2.28GB free)|
|Process Argv||
|Screen Reader|no|
|VM|29%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    2	   152	  5376	code-insiders main
    0	    37	  4352	   gpu-process
    0	   252	  8280	   window (Running Extensions - upfield-edge-solution - Visual Studio Code - Insiders)
    0	   523	  4552	     extensionHost
    0	    37	  5656	       electron_node server.js 
    0	     2	  8596	       cmd /s /c "C:\Users\bhumphreys\.vscode-insiders\extensions\ms-vscode.csharp-1.19.1\.omnisharp\1.32.19\OmniSharp.exe -s c:\Users\bhumphreys\OneDrive\Documents\GitHub\upfield-edge\upfield-edge-solution --hostPID 4552 DotNet:enablePackageRestore=false --encoding utf-8 --loglevel information --plugin C:\Users\bhumphreys\.vscode-insiders\extensions\ms-vscode.csharp-1.19.1\.razor\OmniSharpPlugin\Microsoft.AspNetCore.Razor.OmniSharpPlugin.dll formattingOptions:useTabs=false formattingOptions:tabSize=4 formattingOptions:indentationSize=4"
    0	   127	  7000	         C:\Users\bhumphreys\.vscode-insiders\extensions\ms-vscode.csharp-1.19.1\.omnisharp\1.32.19\OmniSharp.exe  -s c:\Users\bhumphreys\OneDrive\Documents\GitHub\upfield-edge\upfield-edge-solution --hostPID 4552 DotNet:enablePackageRestore=false --encoding utf-8 --loglevel information --plugin C:\Users\bhumphreys\.vscode-insiders\extensions\ms-vscode.csharp-1.19.1\.razor\OmniSharpPlugin\Microsoft.AspNetCore.Razor.OmniSharpPlugin.dll formattingOptions:useTabs=false formattingOptions:tabSize=4 formattingOptions:indentationSize=4
    0	     8	 10668	         console-window-host (Windows internal process)
    0	     3	 27692	       C:\WINDOWS\system32\cmd.exe /d /s /c "docker images --filter "dangling=false" --format "{{.ID}} {{.Repository}} {{.Tag}}""
    0	    13	 22192	         docker  images --filter "dangling=false" --format "{{.ID}} {{.Repository}} {{.Tag}}"
    0	    10	 28800	         console-window-host (Windows internal process)
    0	    28	 28568	       "C:\Program Files\Microsoft VS Code Insiders\Code - Insiders.exe" "c:\Program Files\Microsoft VS Code Insiders\resources\app\extensions\json-language-features\server\dist\jsonServerMain" --node-ipc --clientProcessId=4552
    0	    14	  4836	     watcherService 
    0	     8	 20604	       console-window-host (Windows internal process)
    0	    30	  5028	     searchService
    0	     6	 13464	     winpty-process
    0	    70	 11596	       C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
    0	     9	 15316	       console-window-host (Windows internal process)
    0	     9	 29772	     winpty-process
    0	    10	 20324	       console-window-host (Windows internal process)
    0	    71	 28960	       C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
    0	     4	 12784	         "C:\Python27\Scripts\iotedgehubdev.exe" start -d c:\Users\bhumphreys\OneDrive\Documents\GitHub\upfield-edge\upfield-edge-solution\config\deployment.debug.amd64.json -v
    0	    52	 28376	           "c:\python27\python.exe"  "C:\Python27\Scripts\iotedgehubdev.exe" start -d c:\Users\bhumphreys\OneDrive\Documents\GitHub\upfield-edge\upfield-edge-solution\config\deployment.debug.amd64.json -v
    0	     4	  7048	             docker-compose -f C:\ProgramData\iotedgehubdev\data\data\docker-compose.yml up
    0	    43	 17792	               "c:\python27\python.exe"  "C:\Python27\Scripts\docker-compose.exe" -f C:\ProgramData\iotedgehubdev\data\data\docker-compose.yml up
    0	    75	 15716	   shared-process
    0	    68	 20248	   window (Issue Reporter)
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (Running Extensions - upfield-edge-solution - Visual Studio Code - Insiders)
|    Folder (upfield-edge-solution): 40 files
|      File types: json(15) cache(5) dll(2) debug(2) pdb(2) gitignore(2) cs(2)
|                  csproj(1) windows-amd64(1) arm32v7(1)
|      Conf files: launch.json(1) tasks.json(1) settings.json(1) csproj(1);
```

</details>
<details><summary>Extensions (68)</summary>

Extension|Author (truncated)|Version
---|---|---
azurenodeessentials|azu|0.2.6
vscode-refresh-html|baj|0.0.2
vs-kubernetes|bre|0.0.9
xml2json|bui|1.2.2
gitignore|cod|0.6.0
compulim-vscode-closetag|Com|1.2.0
vscode-markdownlint|Dav|0.26.0
vscode-eslint|dba|1.8.2
xml|Dot|2.4.0
es7-react-js-snippets|dsz|2.2.2
vscode-docker-syntax|dun|0.1.5
vscode-html-css|ecm|0.2.0
vsc-material-theme|Equ|2.8.2
vs-code-xml-format|fab|0.1.5
azure-storage-explorer|for|0.1.2
azure-virtual-machine-explorer|for|0.0.2
docker-explorer|for|0.1.6
iot-extension-pack|for|0.0.5
platformio|for|0.3.0
terminal|for|0.0.10
vscode-node-red|for|0.2.0
vscode-yarn|gam|1.6.0
docker-linter|hen|0.5.0
beautify|Hoo|1.5.0
rest-client|hum|0.21.2
kubernetes-snippets|ipe|0.1.9
csharpextensions|jch|1.3.0
vscode-nuget-package-manager|jmr|1.1.6
docthis|joe|0.7.1
docomment|k--|0.1.6
node-module-intellisense|lei|1.5.0
csharpfixformat|Leo|0.0.81
dotnet-core-commands|mat|1.0.6
azure-pipelines|ms-|1.147.2
vscode-azureappservice|ms-|0.14.0
vscode-azurefunctions|ms-|0.17.0
vscode-azurestorage|ms-|0.7.0
vscode-cosmosdb|ms-|0.10.2
vscode-asa|ms-|0.1.0
windowsiot|ms-|0.0.20
mssql|ms-|1.6.0
python|ms-|2019.4.12954
azure-account|ms-|0.8.2
azurecli|ms-|0.4.5
cpptools|ms-|0.23.0
csharp|ms-|1.19.1
Go|ms-|0.10.2
powershell|ms-|1.12.1
vs-keybindings|ms-|0.2.0
vscode-node-azure-pack|ms-|0.0.6
azurerm-vscode-tools|msa|0.6.0
debugger-for-chrome|msj|4.11.3
ant-tree-viewer|ped|0.3.2
vscode-docker|Pet|0.6.2
material-icon-theme|PKi|3.7.0
qub-xml-vscode|qub|1.2.8
es7-react-js-snippets|rod|1.9.3
arm-snippets|sam|1.4.13
yo|sam|0.9.3
avro|str|0.4.0
azure-event-hub-explorer|Sum|0.1.2
html-preview-vscode|tht|0.2.5
usql-vscode-ext|usq|0.2.15
application-insights|Vis|0.4.2
azure-iot-edge|vsc|1.12.0
azure-iot-toolkit|vsc|2.6.0
vscode-arduino|vsc|0.2.25
vscode-iot-workbench|vsc|0.2.6

(1 theme extensions excluded)

</details>

**Comment by joaomoreno** [comment]
Can you share more insights? What makes you say this?

**Comment by BrentHumphreys** [comment]
I was looking at the Running Tasks window and it showed unresponsive.  Everything in VSCode was taking ages to execute.

I actually found that the Docker CE with the setting to share usage information was causing each docker command to take 5 times longer to execute than it should have.  Once I disabled that, VS Code got much faster.  I am working on an IoT Edge project, so it is using Docker in the background.  W

We can close this issue.
